### ISSUE TITLE
Docker build of 4.2-beta fails with GLIBC version mismatch error in the ruby-jemalloc container

### ISSUE THREAD

**Issue by b2cc** [mastodon/mastodon#26820]
### Steps to reproduce the problem

1. Clone the mastodon repository
2. Check out the tag 4.2.0-beta3: `git checkout v4.2.0-beta3`
3. Run docker build: `docker build . -f Dockerfile -t local/mastodon:v4.2.0-beta3`

### Expected behaviour

Docker image is created successfully

### Actual behaviour

Docker build fails

### Detailed description

Docker builds suddenly fail for me with the following error when the first `bundle` commands are run in the build container:
```
/opt/ruby/bin/ruby: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by /opt/ruby/bin/ruby)
/opt/ruby/bin/ruby: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.35' not found (required by /opt/ruby/lib/libruby.so.3.2)
/opt/ruby/bin/ruby: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by /opt/ruby/lib/libruby.so.3.2)
/opt/ruby/bin/ruby: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.32' not found (required by /opt/ruby/lib/libruby.so.3.2)
/opt/ruby/bin/ruby: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by /opt/ruby/lib/libruby.so.3.2)
```
and then the build fails.

I checked the docker container `node:16.20-bullseye-slim`, and it only provides glibc `2.31-13+deb11u6`:
```
# ldd --version
ldd (Debian GLIBC 2.31-13+deb11u6) 2.31
```

Did the `glibc` requirements change in the Ruby container `ghcr.io/moritzheiber/ruby-jemalloc:3.2.2-slim` or am I missing something?

Complete build log:
[docker_build.log](https://github.com/mastodon/mastodon/files/12538791/docker_build.log)

Thanks for your help!

### Mastodon instance

n/a

### Mastodon version

v4.2.0-beta3

**Comment by renchap** [comment]
This is caused by the base docker image used in our `Dockerfile` switching from Debian Bullseye to Bookworm unexpectedly.

This has been fixed by https://github.com/mastodon/mastodon/pull/26803 and you should be able to apply those change to your Dockerfile and build it successfully.

We will make further changes in our Dockerfile to ensure this does not happen again.

**Comment by b2cc** [comment]
@renchap : Thanks for the quick reply!

**Comment by b2cc** [comment]
@renchap : Ok I looked into this commit, but I'm not quite sure how this is exactly connected to this issue... seems to be for an older image for the 3.5 branch?

Anyways, your comment regarding switching to Debian bookworm let me to a possible solution. I switched the node image to `node:16.20-bookworm-slim` (https://hub.docker.com/_/node/tags?page=1&name=16.20-bookworm-slim), which has the newer `libc` version required by the `ruby-jemalloc` image.

At the beginning of the `Dockerfile`, I changed `ARG NODE_VERSION="16.20-bullseye-slim"`. Also some packages have new versions, update the `apt install` section to use: `libicu72`, `libidn12` and `libssl3`.

This at least makes the image build work again.
